ProcessMaker/Listeners/UserSession.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace ProcessMaker\Listeners;
+
+use Illuminate\Database\Eloquent\Builder;
+use ProcessMaker\Models\UserSession as Session;
+use Illuminate\Support\Str;
+use Jenssegers\Agent\Agent;
+use ProcessMaker\Models\Setting;
+
+class UserSession
+{
+    /**
+     * Create the event listener.
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Handle the event.
+     */
+    public function handle(object $event): void
+    {
+        $agent = new Agent();
+        $user = $event->user;
+
+        $configIP = Setting::configByKey('session-control.ip_restriction');
+        $configDevice = Setting::configByKey('session-control.device_restriction');
+
+        // if block session by IP or Device is enabled, then mark all active sessions as inactive
+        if ($configDevice === '1') {
+            $user->sessions()
+                ->where('is_active', true)
+                ->where([
+                    ['device_name', $agent->device()],
+                    ['device_type', $agent->deviceType()],
+                    ['device_platform', $agent->platform()],
+                ])
+                ->update(['is_active' => false]);
+        }
+        // if kill session by IP or Device is enabled, then kill all active sessions
+        if ($configIP === '2') {
+            $user->sessions()
+                ->where('is_active', true)
+                ->update(['expired_date' => now()]);
```

Comment:
Seems here is missing the filter by IP. Without the filter it will close all the ACTIVE session no matter if it was started from a different IP address.

Reply:
The IP validation was added